src/main/java/core/basesyntax/Square.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax;
+
+public class Square extends Figure {
+    private double side;
+
+    public Square(String color, double side) {
+        super(color);
+        this.side = side;
+    }
+
+    @Override
+    public double getArea() {
+        return Math.pow(side, 2);
+    }
+
+    @Override
+    public void draw() {
+        String randomColor = new ColorSupplier().getRandomColor();
```

Comment:
with the current implementation, you instantiate class ColorSupplier each time the method is being invoked.

think about instantiating ColorSupplier once and reusing it multiple times.